app/src/main/java/unipd/se18/ocrcamera/BrightnessRecognition.java

```diff
@@ -0,0 +1,93 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.Color;
+import android.util.Log;
+import android.widget.Toast;
+import android.os.Handler;
+
+/**
+ * This class is used to calculate the brightness of an image in Bitmap format
+ * @author Pietro Balzan
+ */
+public class BrightnessRecognition {
+    Handler handler;             // handler for communication with the ui thread
+    private Context appContext;  //context of the activity that instances the object of this class
+    private final Bitmap image;  // the bitmap containing the image to analyze
+    private static final String TAG = "BrightnessRecognition";
+
+
+    /**
+     * Constructor of a BrightnessRecognition object containing a Bitmap
+     * @param bmp the bitmap of the image to calculate the brightness of
+     * @param cntxt context of the activity that instances an object of this class
+     * @param hndlr handler for communication with ui thread
+     * @modify image with the new bitmap
+     * @author Pietro Balzan
+     */
+
+    public BrightnessRecognition(Bitmap bmp, Context cntxt, Handler hndlr){
+        appContext= cntxt;
+        image = bmp;
+        handler= hndlr;
+    }
+
+    /**
+     * This method calculates the brightness of the image and prompts a Toast to the user
+     * @param upperBound the toast will be launched if brightness exceeds this value.
+     * @param lowerBound the toast will be launched if brightness is lower than this value.
+     * @param pixelSkip how many pixels to skip each checked pixel. Higher values result in better
+     *                     performance, but a more rough estimate. When pixelSpacing = 1, the method
+     *                     actually calculates the real average brightness, not an estimate.
+     *                     Do not use values for pixelSpacing that are smaller than 1.
+     * @author Pietro Balzan - some changes by Francesco Pham
+     */
+
+    public void imgBrightness(int upperBound, int lowerBound, int pixelSkip){
```

Comment:
Could be useful if this method returns an int for a possible future automatic processing of the picture? For example -1 too dark, 0 ok, 1 too bright?
Is really useful setting externally these parameters? Or they could be calculated automatically or anyway internally to this method?

Reply:
good idea, I will add that feature